### ISSUE TITLE
OpenDDSTpd.dll  failed,Acceptor.cpp(262,1): fatal error C1001: Internal compi ler error

### ISSUE THREAD

**Issue by wuzhuqingfeng** [OpenDDS/OpenDDS#4228]
use version:   ACE+TAO-2.2a_p23_complete.tar     OpenDDS-3.20 

Compiled with ACE+TAO-7.1.0.tar the same problem，please  help.

```
“E:\_proj\_opendds\OpenDDS-3.20\DDS_no_tests.sln”(默认目标) (1) ->
       “E:\_proj\_opendds\OpenDDS-3.20\DevGuideExamples\DCPS\Messenger.minimal\MessengerMinimal_Pub
       lisher.vcxproj.metaproj”(默认目标) (5) ->
       “E:\_proj\_opendds\OpenDDS-3.20\dds\DCPS\InfoRepoDiscovery\OpenDDS_InfoRepoDiscovery.vcxproj
       .metaproj”(默认目标) (6) ->
       “E:\_proj\_opendds\OpenDDS-3.20\dds\DCPS\transport\tcp\OpenDDS_Tcp.vcxproj.metaproj”(默认目标) (
       7) ->
       “E:\_proj\_opendds\OpenDDS-3.20\dds\DCPS\transport\tcp\OpenDDS_Tcp.vcxproj”(默认目标) (45) ->
       (ClCompile 目标) ->
         E:\_proj\_opendds\ACE_wrappers\ace\Acceptor.cpp(262,1): fatal  error C1001: Internal compi
       ler error. [E:\_proj\_opendds\OpenDDS-3.20\dds\DCPS\transport\tcp\OpenDDS_Tcp.vcxproj]

    0 个警告
    1 个错误
 ```

**Comment by iguessthislldo** [comment]
This is [a bug in Visual Studio](https://developercommunity.visualstudio.com/t/176-ICE-for-using-CObject::operator-del/10367118) unfortunately. If they still haven't fixed it in the latest version, you'll have to revert to using VS 17.5 or earlier if you need the tcp transport or else avoid building the tcp transport.

**Comment by mitza-oci** [comment]
This is fixed in Visual Studio 2022 Version 17.7.0
